manifests/init.pp

```diff
@@ -1,38 +1,27 @@
-# == Define: autofs
+# Class: autofs
 
-class autofs(
-  $config_file_owner  = $autofs::params::config_file_owner,
-  $config_file_group  = $autofs::params::config_file_group,
-  $master_config      = $autofs::params::master_config,
-  $map_config_dir     = $autofs::params::map_config_dir,
-  $package_manage     = $autofs::params::package_manage,
-  $package_name       = $autofs::params::package_name,
-  $package_ensure     = $autofs::params::package_ensure,
-  $service_name       = $autofs::params::service_name,
-  $service_manage     = $autofs::params::service_manage,
-  $service_hasrestart = $autofs::params::service_hasrestart,
-  $service_hasstatus  = $autofs::params::service_hasstatus,
-  $service_ensure     = $autofs::params::service_ensure,
-  $service_enable     = $autofs::params::service_enable,
-  $service_restart    = $autofs::params::service_restart,
-  $mapfiles           = {},
-  $mounts             = {},
-  $includes           = {},
+class autofs (
+  String $config_file_owner         = $autofs::params::config_file_owner,
+  String $config_file_group         = $autofs::params::config_file_group,
+  String $master_config             = $autofs::params::master_config,
+  Stdlib::Absolutepath
+  $map_config_dir                   = $autofs::params::map_config_dir,
+  Boolean $package_manage           = $autofs::params::package_manage,
+  Array[String] $package_name       = $autofs::params::package_name,
+  Enum[present, absent, purged, installed]
```

Comment:
String should be good enough, because ensure allows to define a specific version.